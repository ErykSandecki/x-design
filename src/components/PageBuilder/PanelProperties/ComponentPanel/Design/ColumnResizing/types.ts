export type TFocusElement = 'height' | 'width' | '';
