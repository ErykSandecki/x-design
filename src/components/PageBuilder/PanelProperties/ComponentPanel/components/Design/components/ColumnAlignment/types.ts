// types
import { AlignmentHorizontal, AlignmentVertical } from 'types';
import { TButtonGroup } from 'shared/UITools/ButtonGroup/types';

export type TAligmentHorizontalButtonGroup = Array<
  Pick<TButtonGroup, 'name' | 'tooltip'> & { key: AlignmentHorizontal }
>;
export type TAligmentVerticalButtonGoup = Array<Pick<TButtonGroup, 'name' | 'tooltip'> & { key: AlignmentVertical }>;
