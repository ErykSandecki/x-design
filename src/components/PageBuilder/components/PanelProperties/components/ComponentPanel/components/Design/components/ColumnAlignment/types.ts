// types
import { AlignmentHorizontal, AlignmentVertical } from 'types';
import { TButtonGroup } from 'shared/UITools/components/ButtonGroup/types';

export type TAligmentHorizontalButtonGroup = Array<Pick<TButtonGroup, 'name'> & { key: AlignmentHorizontal }>;
export type TAligmentVerticalButtonGoup = Array<Pick<TButtonGroup, 'name'> & { key: AlignmentVertical }>;
