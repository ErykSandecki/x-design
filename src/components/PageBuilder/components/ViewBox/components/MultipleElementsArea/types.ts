// types
import { TElement } from 'types';

export type TCoordinatesData = {
  elementsCoordinates: Array<{
    coordinates: TRectCoordinates;
    id: TElement['id'];
  }>;
  outline: TRectCoordinates;
};
