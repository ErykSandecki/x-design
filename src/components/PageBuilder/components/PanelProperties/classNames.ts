// utils
import { composeClassNames } from 'utils';

export const className = 'PanelProperties';

export const classNames = composeClassNames(
  className,
  [className] as const,
  ['areaHandleResize'] as const,
  ['sections'] as const,
);
