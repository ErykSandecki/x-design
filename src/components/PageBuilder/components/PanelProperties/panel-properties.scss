@import '../../../../styles/mixins/scrollbar';

.PanelProperties {
  &__area-handle-resize {
    cursor: ew-resize;
  }

  &__sections {
    @include scrollbar-styles;

    max-height: calc(100% - 72.5px);
    min-height: calc(100% - 72.5px);
    overflow-y: overlay;
  }
}
