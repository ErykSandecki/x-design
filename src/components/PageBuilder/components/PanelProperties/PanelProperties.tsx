import { FC, memo, useRef, useState } from 'react';
import { useSelector } from 'react-redux';
import { useTranslation } from 'react-i18next';

// components
import ComponentPanel from './components/ComponentPanel/ComponentPanel';
import MainPanel from './components/MainPanel/MainPanel';
import { Box, E2EDataAttribute, TUITypes, UITools } from 'shared';

// hooks
import { useResizeHandler, useTheme } from 'hooks';

// others
import { className, classNames } from './classNames';
import { TABS, translationNameSpace } from './constants';
import { TOOLBAR_HEIGHT } from '../../constants';

// store
import { selectedElementsSelector } from 'store/pageBuilder/selectors';

// styles
import styles from './panel-properties.scss';

// types
import { E2EAttribute, ZIndex } from 'types';
import { Tab } from './enums';

export type TPanelPropertiesProps = {};

const PanelProperties: FC<TPanelPropertiesProps> = () => {
  const boxRef = useRef(null);
  const selectedElements = useSelector(selectedElementsSelector);
  const { t } = useTranslation();
  const [activeTab, setActiveTab] = useState(Tab.design);
  const { classNamesWithTheme, cx } = useTheme(classNames, styles);
  const { onMouseDownX, width } = useResizeHandler(
    0,
    240,
    window.innerHeight - TOOLBAR_HEIGHT,
    500,
    0,
    240,
    boxRef,
  );

  return (
    <Box
      classes={{ className: cx(classNamesWithTheme[className]) }}
      e2eValue="panel-properties"
      onMouseDown={(event) => event.stopPropagation()}
      onKeyDown={(event) => event.stopPropagation()}
      ref={boxRef}
      style={{ touchAction: 'manipulation', width }}
      sx={{
        bg: 'neutral5',
        borderLeft: 1,
        boxSizing: 'border-box',
        height: `calc(100vh - ${TOOLBAR_HEIGHT}px)`,
        position: 'absolute',
        right: 0,
        top: `${TOOLBAR_HEIGHT}px`,
        zIndex: ZIndex.standard,
      }}
    >
      <E2EDataAttribute type={E2EAttribute.resize}>
        <div
          className={cx(classNamesWithTheme.areaHandleResize)}
          onMouseDown={(event) => onMouseDownX(event, true)}
        />
      </E2EDataAttribute>
      <UITools.Section label={t(`${translationNameSpace}.section.label`)}>
        <UITools.Tabs
          activeTab={activeTab}
          setActiveTab={setActiveTab as TUITypes['TTabsProps']['setActiveTab']}
          tabs={TABS}
        />
      </UITools.Section>
      <div className={cx(classNamesWithTheme.sections)}>
        {selectedElements.length === 0 ? (
          <MainPanel activeTab={activeTab} />
        ) : (
          <ComponentPanel activeTab={activeTab} />
        )}
      </div>
    </Box>
  );
};

export default memo(PanelProperties);
