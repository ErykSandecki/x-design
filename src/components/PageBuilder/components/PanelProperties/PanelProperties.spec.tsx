import { fireEvent, render } from '@testing-library/react';
import { Provider } from 'react-redux';

// components
import PanelProperties from './PanelProperties';

// mocks
import {
  elementAllDataMock,
  elementDynamicDataMock,
  elementStaticDataMock,
  pageBuilderStateMock,
  selectedElementMock,
} from 'test/mocks/reducer/pageBuilderMock';
import 'test/mocks/sagas/allSagas';

// others
import { REDUCER_KEY as PAGE_BUILDER } from 'store/pageBuilder/actionsType';

// store
import { configureStore } from 'store/store';

// types
import { E2EAttribute, MouseButton } from 'types';

// utils
import { getByE2EAttribute } from 'test';

const mockCallBack = jest.fn();

const currentPage = pageBuilderStateMock[PAGE_BUILDER].pages['0'];
const stateMock = {
  ...pageBuilderStateMock,
};

describe('PanelProperties snapshots', () => {
  it('should render PanelProperties', () => {
    // mock
    const store = configureStore();

    // before
    const { asFragment } = render(
      <Provider store={store}>
        <PanelProperties />
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render when some element is selected', () => {
    // mock
    const store = configureStore({
      ...stateMock,
      [PAGE_BUILDER]: {
        ...stateMock[PAGE_BUILDER],
        ...pageBuilderStateMock[PAGE_BUILDER],
        pages: {
          ...pageBuilderStateMock[PAGE_BUILDER].pages,
          ['0']: {
            ...currentPage,
            elements: {
              allData: {
                ['-1']: {
                  ...currentPage.elements.allData['-1'],
                  children: [elementAllDataMock.id],
                },
                [elementAllDataMock.id]: {
                  ...elementAllDataMock,
                },
              },
              dynamicData: {
                ['-1']: {
                  ...currentPage.elements.dynamicData['-1'],
                  children: [elementDynamicDataMock.id],
                },
                [elementDynamicDataMock.id]: elementDynamicDataMock,
              },
              staticData: {
                ['-1']: {
                  ...currentPage.elements.staticData['-1'],
                  children: [elementStaticDataMock.id],
                },
                [elementStaticDataMock.id]: {
                  ...elementStaticDataMock,
                },
              },
            },
            selectedElements: [selectedElementMock],
          },
        },
      },
    });

    // before
    const { asFragment } = render(
      <Provider store={store}>
        <PanelProperties />
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render with changed width', () => {
    // mock
    const store = configureStore();

    // before
    const { asFragment, container } = render(
      <Provider store={store}>
        <PanelProperties />
      </Provider>,
    );

    // action
    fireEvent.mouseDown(getByE2EAttribute(container, E2EAttribute.resize), {
      buttons: MouseButton.lmb,
    });
    fireEvent.mouseMove(document, { clientX: 500 });

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('PanelProperties behaviors', () => {
  it('should stop propagation on key down', () => {
    // mock
    const store = configureStore();

    // before
    const { container } = render(
      <Provider store={store}>
        <div onKeyDown={mockCallBack}>
          <PanelProperties />
        </div>
      </Provider>,
    );

    // action
    fireEvent.keyDown(getByE2EAttribute(container, E2EAttribute.box, 'panel-properties'));

    // result
    expect(mockCallBack.mock.calls.length).toBe(0);
  });
});
