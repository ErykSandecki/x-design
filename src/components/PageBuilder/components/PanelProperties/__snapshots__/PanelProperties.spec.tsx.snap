// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelProperties snapshots should render PanelProperties 1`] = `
<DocumentFragment>
  <div
    class="css-1nlc499 PanelProperties PanelProperties--light"
    data-test-box="panel-properties"
    style="width: 240px;"
  >
    <div
      class="PanelProperties__area-handle-resize PanelProperties__area-handle-resize--light"
      data-test-resize=""
    />
    <div
      class="css-wcnq0t Section Section--light"
      data-test-section=""
    >
      <div
        class="Section__header Section__header--light"
      >
        <small
          class="css-wcnq0t Section__label Section__label--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
          data-test-text=""
          style="text-align: inherit;"
        >
          Mode
        </small>
        <div
          class="Section__buttons Section__buttons--light"
        />
      </div>
      <div
        class="css-wcnq0t Tabs Tabs--light"
        data-test-box=""
      >
        <div
          class="Tabs__tab Tabs__tab--light Tabs__tab--active Tabs__tab--active--light"
          data-test-tab="design"
        >
          Design
        </div>
        <div
          class="Tabs__tab Tabs__tab--light"
          data-test-tab="prototype\`,"
        >
          Prototype
        </div>
      </div>
    </div>
    <div
      class="PanelProperties__sections PanelProperties__sections--light"
    >
      <div
        class="css-wcnq0t Section Section--light"
        data-test-section=""
      >
        <div
          class="Section__header Section__header--light"
        >
          <small
            class="css-wcnq0t Section__label Section__label--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
            data-test-text=""
            style="text-align: inherit;"
          >
            Page
          </small>
          <div
            class="Section__buttons Section__buttons--light"
          />
        </div>
        <div
          class="css-wcnq0t SectionColumn SectionColumn--light"
          data-test-section=""
        >
          <div
            class="SectionColumn__labels SectionColumn__labels--light"
            style="width: calc(100% - 24px - 0px - 8px);"
          />
          <div
            class="SectionColumn__wrapper SectionColumn__wrapper--light"
          >
            <div
              class="SectionColumn__content SectionColumn__content--light SectionColumn__content--single SectionColumn__content--single--light"
              style="width: calc(100% - 24px - 0px - 8px);"
            >
              <div
                class="css-wcnq0t FieldGroup FieldGroup--light"
                data-test-field-group=""
              >
                <div
                  class="css-wcnq0t ColorPicker__hex-input ColorPicker__hex-input--light TextField TextField--light"
                  data-test-text-field="color"
                >
                  <div
                    aria-describedby="test-id"
                  >
                    <div
                      class="css-wcnq0t Color Color--light"
                      data-test-color=""
                    >
                      <div
                        class="Color__picker Color__picker--light"
                        style="background-color: rgb(241, 242, 246);"
                      />
                      <div
                        class="Color__picker-alpha Color__picker-alpha--light"
                        style="background-color: rgb(241, 242, 246);"
                      />
                      <div
                        class="Color__picker-texture Color__picker-texture--light"
                      />
                    </div>
                  </div>
                  <input
                    class="TextField__input TextField__input--light"
                    data-test-text-field-input="color"
                    maxlength="6"
                    value="f1f2f6"
                  />
                </div>
                <div
                  class="css-wcnq0t ColorPicker__alpha-input ColorPicker__alpha-input--light TextField TextField--light"
                  data-test-text-field="alpha"
                >
                  <input
                    class="TextField__input TextField__input--light"
                    data-test-text-field-input="alpha"
                    max="100"
                    maxlength="6"
                    min="0"
                    type="number"
                    value="100"
                  />
                  <div
                    class="ScrubbableInput ScrubbableInput--light"
                    data-test-scrubbable-input=""
                  >
                    %
                  </div>
                </div>
              </div>
            </div>
            <div
              class="SectionColumn__buttons SectionColumn__buttons--light"
            >
              <div
                class="css-wcnq0t ButtonIcon ButtonIcon--light"
                data-test-button-icon=""
              >
                <div
                  class="css-wcnq0t Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                  data-test-icon="eyes-opened"
                  height="14"
                  name="EyesOpened"
                  width="14"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PanelProperties snapshots should render when some element is selected 1`] = `
<DocumentFragment>
  <div
    class="css-1nlc499 PanelProperties PanelProperties--light"
    data-test-box="panel-properties"
    style="width: 240px;"
  >
    <div
      class="PanelProperties__area-handle-resize PanelProperties__area-handle-resize--light"
      data-test-resize=""
    />
    <div
      class="css-wcnq0t Section Section--light"
      data-test-section=""
    >
      <div
        class="Section__header Section__header--light"
      >
        <small
          class="css-wcnq0t Section__label Section__label--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
          data-test-text=""
          style="text-align: inherit;"
        >
          Mode
        </small>
        <div
          class="Section__buttons Section__buttons--light"
        />
      </div>
      <div
        class="css-wcnq0t Tabs Tabs--light"
        data-test-box=""
      >
        <div
          class="Tabs__tab Tabs__tab--light Tabs__tab--active Tabs__tab--active--light"
          data-test-tab="design"
        >
          Design
        </div>
        <div
          class="Tabs__tab Tabs__tab--light"
          data-test-tab="prototype\`,"
        >
          Prototype
        </div>
      </div>
    </div>
    <div
      class="PanelProperties__sections PanelProperties__sections--light"
    >
      <div
        class="css-wcnq0t Section Section--light"
        data-test-section=""
      >
        <div
          class="css-wcnq0t SectionColumn SectionColumn--light"
          data-test-section=""
        >
          <div
            class="SectionColumn__labels SectionColumn__labels--light"
            style="width: calc(100% - 48px - 2.5px - 8px);"
          />
          <div
            class="SectionColumn__wrapper SectionColumn__wrapper--light"
          >
            <div
              class="SectionColumn__content SectionColumn__content--light SectionColumn__content--single SectionColumn__content--single--light"
              style="width: calc(100% - 48px - 2.5px - 8px);"
            >
              <small
                class="css-wcnq0t Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
                data-test-text=""
                style="text-align: inherit;"
              >
                Frame
              </small>
            </div>
            <div
              class="SectionColumn__buttons SectionColumn__buttons--light"
            >
              <div
                class="css-wcnq0t ButtonIcon ButtonIcon--light"
                data-test-button-icon=""
              >
                <div
                  class="css-wcnq0t Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                  data-test-icon="html-tag"
                  height="14"
                  name="HtmlTag"
                  width="14"
                />
              </div>
              <div
                class="css-wcnq0t ButtonIcon ButtonIcon--light"
                data-test-button-icon=""
              >
                <div
                  class="css-wcnq0t Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                  data-test-icon="component"
                  height="14"
                  name="Component"
                  width="14"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="css-wcnq0t Section Section--light"
        data-test-section=""
      >
        <div
          class="Section__header Section__header--light"
        >
          <small
            class="css-wcnq0t Section__label Section__label--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
            data-test-text=""
            style="text-align: inherit;"
          >
            Position
          </small>
          <div
            class="Section__buttons Section__buttons--light"
          />
        </div>
        <div
          class="css-wcnq0t SectionColumn SectionColumn--light SectionColumn--with-margin SectionColumn--with-margin--light"
          data-test-section=""
        >
          <div
            class="SectionColumn__labels SectionColumn__labels--light"
            style="width: calc(100% - 24px - 0px - 8px);"
          >
            <small
              class="css-wcnq0t SectionColumn__label SectionColumn__label--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
              data-test-text=""
              style="text-align: inherit;"
            >
              Alignment
            </small>
          </div>
          <div
            class="SectionColumn__wrapper SectionColumn__wrapper--light"
          >
            <div
              class="SectionColumn__content SectionColumn__content--light SectionColumn__content--two-inputs SectionColumn__content--two-inputs--light"
              style="width: calc(100% - 24px - 0px - 8px);"
            >
              <div
                class="css-wcnq0t ButtonGroup ButtonGroup--light"
                data-test-button-group="horizontal-alignment"
              >
                <div
                  class="ButtonGroup__button ButtonGroup__button--light ButtonGroup__button--disabled ButtonGroup__button--disabled--light"
                  data-test-button-group-input="align-horizontal-left"
                >
                  <div
                    class="css-wcnq0t ButtonGroup__icon ButtonGroup__icon--light ButtonGroup__icon--disabled ButtonGroup__icon--disabled--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                    data-test-icon="align-horizontal-left"
                    height="12"
                    name="AlignHorizontalLeft"
                    width="12"
                  />
                </div>
                <div
                  class="ButtonGroup__button ButtonGroup__button--light ButtonGroup__button--disabled ButtonGroup__button--disabled--light"
                  data-test-button-group-input="align-horizontal-center"
                >
                  <div
                    class="css-wcnq0t ButtonGroup__icon ButtonGroup__icon--light ButtonGroup__icon--disabled ButtonGroup__icon--disabled--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                    data-test-icon="align-horizontal-center"
                    height="12"
                    name="AlignHorizontalCenter"
                    width="12"
                  />
                </div>
                <div
                  class="ButtonGroup__button ButtonGroup__button--light ButtonGroup__button--disabled ButtonGroup__button--disabled--light"
                  data-test-button-group-input="align-horizontal-right"
                >
                  <div
                    class="css-wcnq0t ButtonGroup__icon ButtonGroup__icon--light ButtonGroup__icon--disabled ButtonGroup__icon--disabled--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                    data-test-icon="align-horizontal-right"
                    height="12"
                    name="AlignHorizontalRight"
                    width="12"
                  />
                </div>
              </div>
              <div
                class="css-wcnq0t ButtonGroup ButtonGroup--light"
                data-test-button-group="vertical-alignment"
              >
                <div
                  class="ButtonGroup__button ButtonGroup__button--light ButtonGroup__button--disabled ButtonGroup__button--disabled--light"
                  data-test-button-group-input="align-vertical-top"
                >
                  <div
                    class="css-wcnq0t ButtonGroup__icon ButtonGroup__icon--light ButtonGroup__icon--disabled ButtonGroup__icon--disabled--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                    data-test-icon="align-vertical-top"
                    height="12"
                    name="AlignVerticalTop"
                    width="12"
                  />
                </div>
                <div
                  class="ButtonGroup__button ButtonGroup__button--light ButtonGroup__button--disabled ButtonGroup__button--disabled--light"
                  data-test-button-group-input="align-vertical-center"
                >
                  <div
                    class="css-wcnq0t ButtonGroup__icon ButtonGroup__icon--light ButtonGroup__icon--disabled ButtonGroup__icon--disabled--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                    data-test-icon="align-vertical-center"
                    height="12"
                    name="AlignVerticalCenter"
                    width="12"
                  />
                </div>
                <div
                  class="ButtonGroup__button ButtonGroup__button--light ButtonGroup__button--disabled ButtonGroup__button--disabled--light"
                  data-test-button-group-input="align-vertical-bottom"
                >
                  <div
                    class="css-wcnq0t ButtonGroup__icon ButtonGroup__icon--light ButtonGroup__icon--disabled ButtonGroup__icon--disabled--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                    data-test-icon="align-vertical-bottom"
                    height="12"
                    name="AlignVerticalBottom"
                    width="12"
                  />
                </div>
              </div>
            </div>
            <div
              class="SectionColumn__buttons SectionColumn__buttons--light"
            />
          </div>
        </div>
        <div
          class="css-wcnq0t SectionColumn SectionColumn--light SectionColumn--with-margin SectionColumn--with-margin--light"
          data-test-section=""
        >
          <div
            class="SectionColumn__labels SectionColumn__labels--light"
            style="width: calc(100% - 24px - 0px - 8px);"
          >
            <small
              class="css-wcnq0t SectionColumn__label SectionColumn__label--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
              data-test-text=""
              style="text-align: inherit;"
            >
              Position
            </small>
          </div>
          <div
            class="SectionColumn__wrapper SectionColumn__wrapper--light"
          >
            <div
              class="SectionColumn__content SectionColumn__content--light SectionColumn__content--two-inputs SectionColumn__content--two-inputs--light"
              style="width: calc(100% - 24px - 0px - 8px);"
            >
              <div
                class="css-wcnq0t TextField TextField--light TextField--full-width TextField--full-width--light"
                data-test-text-field="x"
              >
                <div
                  class="ScrubbableInput ScrubbableInput--light"
                  data-test-scrubbable-input="x"
                >
                  <small
                    class="css-wcnq0t Typography Typography--light Typography--neutral2 Typography--neutral2--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
                    data-test-text=""
                    style="text-align: inherit;"
                  >
                    X
                  </small>
                </div>
                <input
                  class="TextField__input TextField__input--light"
                  data-test-text-field-input="x"
                  maxlength="6"
                  type="number"
                  value="0"
                />
              </div>
              <div
                class="css-wcnq0t TextField TextField--light TextField--full-width TextField--full-width--light"
                data-test-text-field="y"
              >
                <div
                  class="ScrubbableInput ScrubbableInput--light"
                  data-test-scrubbable-input="y"
                >
                  <small
                    class="css-wcnq0t Typography Typography--light Typography--neutral2 Typography--neutral2--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
                    data-test-text=""
                    style="text-align: inherit;"
                  >
                    Y
                  </small>
                </div>
                <input
                  class="TextField__input TextField__input--light"
                  data-test-text-field-input="y"
                  maxlength="6"
                  type="number"
                  value="0"
                />
              </div>
            </div>
            <div
              class="SectionColumn__buttons SectionColumn__buttons--light"
            />
          </div>
        </div>
        <div
          class="css-wcnq0t SectionColumn SectionColumn--light"
          data-test-section=""
        >
          <div
            class="SectionColumn__labels SectionColumn__labels--light"
            style="width: calc(100% - 24px - 0px - 8px);"
          >
            <small
              class="css-wcnq0t SectionColumn__label SectionColumn__label--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
              data-test-text=""
              style="text-align: inherit;"
            >
              Rotation
            </small>
          </div>
          <div
            class="SectionColumn__wrapper SectionColumn__wrapper--light"
          >
            <div
              class="SectionColumn__content SectionColumn__content--light SectionColumn__content--two-inputs SectionColumn__content--two-inputs--light"
              style="width: calc(100% - 24px - 0px - 8px);"
            >
              <div
                class="css-wcnq0t TextField TextField--light TextField--full-width TextField--full-width--light"
                data-test-text-field="angle"
              >
                <div
                  class="ScrubbableInput ScrubbableInput--light"
                  data-test-scrubbable-input="angle"
                >
                  <div
                    class="css-wcnq0t ScrubbableInput__icon ScrubbableInput__icon--light Icon Icon--light Icon--neutral2 Icon--neutral2--light"
                    data-test-icon="protractor"
                    height="16"
                    name="Protractor"
                    width="16"
                  />
                </div>
                <input
                  class="TextField__input TextField__input--light"
                  data-test-text-field-input="angle"
                  maxlength="6"
                  type="number"
                  value="0"
                />
              </div>
              <div
                class="css-wcnq0t ButtonGroup ButtonGroup--light ButtonGroup--full-width ButtonGroup--full-width--light"
                data-test-button-group="layout-position"
              >
                <div
                  class="ButtonGroup__button ButtonGroup__button--light"
                  data-test-button-group-input="toggle-rotate"
                >
                  <div
                    class="css-wcnq0t ButtonGroup__icon ButtonGroup__icon--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                    data-test-icon="toggle-rotate"
                    height="12"
                    name="ToggleRotate"
                    width="12"
                  />
                </div>
                <div
                  class="ButtonGroup__button ButtonGroup__button--light"
                  data-test-button-group-input="flip-horizontal"
                >
                  <div
                    class="css-wcnq0t ButtonGroup__icon ButtonGroup__icon--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                    data-test-icon="flip-horizontal"
                    height="12"
                    name="FlipHorizontal"
                    width="12"
                  />
                </div>
                <div
                  class="ButtonGroup__button ButtonGroup__button--light"
                  data-test-button-group-input="flip-vertical"
                >
                  <div
                    class="css-wcnq0t ButtonGroup__icon ButtonGroup__icon--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                    data-test-icon="flip-vertical"
                    height="12"
                    name="FlipVertical"
                    width="12"
                  />
                </div>
              </div>
            </div>
            <div
              class="SectionColumn__buttons SectionColumn__buttons--light"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PanelProperties snapshots should render with changed width 1`] = `
<DocumentFragment>
  <div
    class="css-1nlc499 PanelProperties PanelProperties--light"
    data-test-box="panel-properties"
    style="width: 500px;"
  >
    <div
      class="PanelProperties__area-handle-resize PanelProperties__area-handle-resize--light"
      data-test-resize=""
    />
    <div
      class="css-wcnq0t Section Section--light"
      data-test-section=""
    >
      <div
        class="Section__header Section__header--light"
      >
        <small
          class="css-wcnq0t Section__label Section__label--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
          data-test-text=""
          style="text-align: inherit;"
        >
          Mode
        </small>
        <div
          class="Section__buttons Section__buttons--light"
        />
      </div>
      <div
        class="css-wcnq0t Tabs Tabs--light"
        data-test-box=""
      >
        <div
          class="Tabs__tab Tabs__tab--light Tabs__tab--active Tabs__tab--active--light"
          data-test-tab="design"
        >
          Design
        </div>
        <div
          class="Tabs__tab Tabs__tab--light"
          data-test-tab="prototype\`,"
        >
          Prototype
        </div>
      </div>
    </div>
    <div
      class="PanelProperties__sections PanelProperties__sections--light"
    >
      <div
        class="css-wcnq0t Section Section--light"
        data-test-section=""
      >
        <div
          class="Section__header Section__header--light"
        >
          <small
            class="css-wcnq0t Section__label Section__label--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
            data-test-text=""
            style="text-align: inherit;"
          >
            Page
          </small>
          <div
            class="Section__buttons Section__buttons--light"
          />
        </div>
        <div
          class="css-wcnq0t SectionColumn SectionColumn--light"
          data-test-section=""
        >
          <div
            class="SectionColumn__labels SectionColumn__labels--light"
            style="width: calc(100% - 24px - 0px - 8px);"
          />
          <div
            class="SectionColumn__wrapper SectionColumn__wrapper--light"
          >
            <div
              class="SectionColumn__content SectionColumn__content--light SectionColumn__content--single SectionColumn__content--single--light"
              style="width: calc(100% - 24px - 0px - 8px);"
            >
              <div
                class="css-wcnq0t FieldGroup FieldGroup--light"
                data-test-field-group=""
              >
                <div
                  class="css-wcnq0t ColorPicker__hex-input ColorPicker__hex-input--light TextField TextField--light"
                  data-test-text-field="color"
                >
                  <div
                    aria-describedby="test-id"
                  >
                    <div
                      class="css-wcnq0t Color Color--light"
                      data-test-color=""
                    >
                      <div
                        class="Color__picker Color__picker--light"
                        style="background-color: rgb(241, 242, 246);"
                      />
                      <div
                        class="Color__picker-alpha Color__picker-alpha--light"
                        style="background-color: rgb(241, 242, 246);"
                      />
                      <div
                        class="Color__picker-texture Color__picker-texture--light"
                      />
                    </div>
                  </div>
                  <input
                    class="TextField__input TextField__input--light"
                    data-test-text-field-input="color"
                    maxlength="6"
                    value="f1f2f6"
                  />
                </div>
                <div
                  class="css-wcnq0t ColorPicker__alpha-input ColorPicker__alpha-input--light TextField TextField--light"
                  data-test-text-field="alpha"
                >
                  <input
                    class="TextField__input TextField__input--light"
                    data-test-text-field-input="alpha"
                    max="100"
                    maxlength="6"
                    min="0"
                    type="number"
                    value="100"
                  />
                  <div
                    class="ScrubbableInput ScrubbableInput--light"
                    data-test-scrubbable-input=""
                  >
                    %
                  </div>
                </div>
              </div>
            </div>
            <div
              class="SectionColumn__buttons SectionColumn__buttons--light"
            >
              <div
                class="css-wcnq0t ButtonIcon ButtonIcon--light"
                data-test-button-icon=""
              >
                <div
                  class="css-wcnq0t Icon Icon--light Icon--neutral1 Icon--neutral1--light"
                  data-test-icon="eyes-opened"
                  height="14"
                  name="EyesOpened"
                  width="14"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
